### PR TITLE
US887031: Use secure Trust Manager when connecting to RabbitMQ via TLS

### PR DESCRIPTION
**Pull request by jonny-mccoubrey**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=887031

### CODE REVIEW

File: worker-message-prioritization-distribution-container/pom.xml [hunk 1]
```diff
@@ -146,11 +146,38 @@
...
+                                <platform>linux/amd64</platform>
+                            </run>
+                        </image>
 
                         <!-- Run the RabbitMQ image -->
                         <image>
                             <alias>rabbitmq</alias>
-                            <name>${dockerHubPublic}/library/rabbitmq:3-management</name>
+                            <name>${dockerHubPublic}/library/rabbitmq:3-management-tls</name>
+                            <build>
```
Comment (andyreidz) [inline worker-message-prioritization-distribution-container/pom.xml:178]: We don't want this image published

File: worker-message-prioritization-distribution-container/pom.xml [hunk 2]
```diff
@@ -146,11 +146,38 @@
                     <containerNamePattern>%a-%t</containerNamePattern>
                     <verbose>true</verbose>
                     <images>
+                        <image>
```
Comment (andyreidz) [inline worker-message-prioritization-distribution-container/pom.xml:149]: We don't want this image published, check you check we're not pushing it to artifactory?

**Review by andyreidz — CHANGES_REQUESTED** [review]

File: worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/DistributorModule.java [hunk 3]
```diff
@@ -156,13 +162,26 @@ ConnectionFactory provideConnectionFactory(final MessageDistributorConfig messag
         ConnectionFactory connectionFactory = new ConnectionFactory();
 
         try {
+            if (messageDistributorConfig.getRabbitmqProtocol().equalsIgnoreCase("amqps")) {
+                final String tmfAlgorithm = System.getProperty(
+                        TRUST_MANAGER_FACTORY_ALGORITHM, TrustManagerFactory.getDefaultAlgorithm()
+                );
+                final TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(tmfAlgorithm);
+                trustManagerFactory.init((KeyStore) null);
```
Comment (jonny-mccoubrey) [inline worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/DistributorModule.java:170]: @andyreidz the `init()` method throws a `KeyStoreException`

File: worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/DistributorModule.java [hunk 4]
```diff
@@ -156,13 +162,26 @@ ConnectionFactory provideConnectionFactory(final MessageDistributorConfig messag
...
+            }
+
             final URI rabbitUrl = new URI(String.format("%s://%s:%s",
                     messageDistributorConfig.getRabbitmqProtocol(),
                     messageDistributorConfig.getRabbitMQHost(),
                     messageDistributorConfig.getRabbitMQPort()
             ));
             connectionFactory.setUri(rabbitUrl);
-        } catch (final URISyntaxException | NoSuchAlgorithmException | KeyManagementException e) {
+        } catch (final URISyntaxException | NoSuchAlgorithmException | KeyManagementException | KeyStoreException e) {
```
[the remark this answers is not in the document]
Comment (jonny-mccoubrey) [reply worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/DistributorModule.java:184]: Do you mean we should have a separate catch for KeyStoreException, the same as what we are now doing in worker framework?

File: worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/DistributorModule.java (hunk 3 above)
Comment (andyreidz) [reply worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/DistributorModule.java:170]: Sorry, my mistake, you're not using the util library from worker framework here :)

File: worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/DistributorModule.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (andyreidz) [reply worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/DistributorModule.java:184]: Sorry my mistake.

**Review by andyreidz — CHANGES_REQUESTED** [review]
Thanks, looks good.
Don't forget releases notes and mention the new system property.

File: worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/DistributorModule.java [hunk 5]
```diff
@@ -156,13 +162,26 @@ ConnectionFactory provideConnectionFactory(final MessageDistributorConfig messag
         ConnectionFactory connectionFactory = new ConnectionFactory();
 
         try {
+            if (messageDistributorConfig.getRabbitmqProtocol().equalsIgnoreCase("amqps")) {
+                final String tmfAlgorithm = System.getProperty(
+                        TRUST_MANAGER_FACTORY_ALGORITHM, TrustManagerFactory.getDefaultAlgorithm()
```
Comment (andyreidz) [inline worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/DistributorModule.java:167]: While I was reviewing the release notes I wanted to see what the "default" was. 
I looked at the implementation of TrustManagerFactory.getDefaultAlgorithm() and I see there's already a way to externally configure a value for `ssl.TrustManagerFactory.algorithm` (See Security.java initialize())

Let's remove this additional way.

File: release-notes-2.1.0.md [hunk 6]
```diff
@@ -5,4 +5,9 @@ ${version-number}
 
 #### New Features
 
+#### Bug Fixes
+- **I887031:** Providing a Trust Manager when connecting to RabbitMQ server via TLS. Trust Manager factory
```
Comment (andyreidz) [inline release-notes-2.1.0.md:9]: Reword to

Replace use of the TrustEverythingTrustManager with a TrustManager provided by the javax.net.ssl.TrustManagerFactory.

**Review by andyreidz — CHANGES_REQUESTED** [review]

**Review by andyreidz — APPROVED** [review]
🙌

**Pull request merged**